src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,98 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import {
+  getPostDetails,
+  getPostComments,
+  removeComment,
+  addComment,
+} from '../../api/api';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+type Props = {
+  selectedPostId: number;
+};
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+export const PostDetails: React.FC<Props> = (props) => {
+  const { selectedPostId } = props;
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentNotHidden, setIsCommentNotHidden] = useState(true);
 
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
+  useEffect(() => {
+    (async () => {
+      const [selectedPostFromApi, commentsFromApi] = await Promise.all([
+        await getPostDetails(selectedPostId),
+        await getPostComments(selectedPostId),
+      ]);
 
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
+      setSelectedPost(selectedPostFromApi);
+      setComments(commentsFromApi);
+    })();
+  }, [selectedPostId]);
+
+  if (!selectedPost) {
+    return null;
+  }
```

Comment:
what is the purpose of it?

Reply:
to avoid this situation: [screen](https://clip2net.com/s/4dqhOiE)